compress/compress_test.go

```diff
@@ -88,19 +88,24 @@ func testEncodeDecode(t *testing.T, m kafka.Message, codec pkg.Codec) {
 	t.Run("encode with "+codec.Name(), func(t *testing.T) {
 		r1, err = compress(codec, m.Value)
 		if err != nil {
-			t.Error(err)
+			t.Fatal(err)
 		}
 	})
 
 	t.Run("decode with "+codec.Name(), func(t *testing.T) {
+		if r1 == nil {
```

Comment:
is r1 only nil if the first test fails? in that case does the `t.Fatal` cover this already?

Reply:
`r1` may be `nil` if the first test is not run, which is what this check is intended to cover (for example if `-run` is used in the `go test` invocation).